dbt/adapters/databricks/impl.py

```diff
@@ -260,86 +250,9 @@ def run_sql_for_tests(
-
-        cluster_id = credentials.cluster_id
-        if not cluster_id:
-            raise ValueError("Python model doesn't support SQL Warehouses")
-
-        command_name = f"dbt-databricks_{version}"
-
-        invocation_env = os.environ.get(DBT_DATABRICKS_INVOCATION_ENV)
-        if invocation_env is not None and len(invocation_env) > 0:
-            self.ConnectionManager.validate_invocation_env(invocation_env)
-            command_name = f"{command_name}-{invocation_env}"
-
-        command_name += "-" + str(uuid.uuid1())
-
-        api_client = Api12Client(
-            host=credentials.host, token=cast(str, credentials.token), command_name=command_name
-        )
-
-        try:
-            # Create an execution context
-            context_id = api_client.Context.create(cluster_id)
-
-            try:
-                # Run a command
-                command_id = api_client.Command.execute(
-                    cluster_id=cluster_id,
-                    context_id=context_id,
-                    command=compiled_code,
-                )
-
-                # poll until job finish
-                status: str
-                start = time.time()
-                exceeded_timeout = False
-                terminal_states = ("Cancelled", "Error", "Finished")
-                while True:
-                    response = api_client.Command.status(
-                        cluster_id=cluster_id, context_id=context_id, command_id=command_id
-                    )
-                    status = response["status"]
-                    if status in terminal_states:
-                        break
-                    if time.time() - start > timeout:
-                        exceeded_timeout = True
-                        break
-                    time.sleep(3)
-                if exceeded_timeout:
-                    raise dbt.exceptions.RuntimeException("python model run timed out")
-                if status != "Finished":
-                    raise dbt.exceptions.RuntimeException(
-                        "python model run ended in state "
-                        f"{status} with state_message\n{response['results']['data']}"
-                    )
-                if response["results"]["resultType"] == "error":
-                    raise dbt.exceptions.RuntimeException(
-                        f"Python model failed with traceback as:\n{response['results']['cause']}"
-                    )
-
-                return self.connections.get_response(None)
-
-            finally:
-                # Delete the execution context
-                api_client.Context.destroy(cluster_id=cluster_id, context_id=context_id)
-        except HTTPError as e:
-            raise dbt.exceptions.RuntimeException(str(e)) from e
-        finally:
-            api_client.close()
+    @property
+    def python_submission_helpers(self) -> Dict[str, Type[PythonJobHelper]]:
+        return {"commands": CommandApiPythonJobHelper}
```

Comment:
Hey @ueshin  sorry for updating things so frequent, but after talking to @lostmygithubaccount we decided to update the submission method to reflect the cluster type that we run python model on instead of the API we use.
Here's the [change on the dbt-spark](https://github.com/dbt-labs/dbt-spark/pull/467/files#diff-af2c2165403b748d8057a7770f80e17d100d0cf8999c4160d57b634ee15d4977R385) side.

I think for here all we need to do is change `commands` to `all_purpose_cluster ` 
```suggestion
        return {"all_purpose_cluster": CommandApiPythonJobHelper}
```

Reply:
oh, and maybe `check_credentials` was changed.

Thanks for letting me know. I'll handle it.

Reply:
Made minimum changes. I'll submit a follow-up PR.